TestClass.cpp

```diff
@@ -0,0 +1,234 @@
+#include <iostream>
+#include <cmath>
+#include <cassert>  
+
+using namespace std; //use std::cout
+
+#define FILE (__FILE__)
+#define LINE (__LINE__)
+
+// #define LOG_LENGTH (300)
+const int LOG_LENGTH = 300;
+
+void print_log(const char *filename,
+               int         line,
+               const char *message)
+{
+    printf("F:%s L:%d : %s\n", filename, line, message);
+}
+
+class TestClass
+{
+    private:
+        int var1;
+        int var2;
+        // double var3;
+        // double var4;
+        int max_average_value;
+        int max_square_root_value;
+
+    public:
+    //get and set functions are public
+
+        TestClass(int a, int b)
+        {
+            var1 = a;
+            var2 = b;
+            max_average_value = 10;
+            max_square_root_value = 7;
+        }
+
+#if 0
+        // do not create constructors with same number of parameters, ideally different number of parameters
+        TestClass(double a, double b)
+        {
+            var3 = a;
+            var4 = b;
+        }
+
+        double getSum(double var1, double var2)
+        {
+            return (var3 + var4);
+        }
+#endif        
+
+        int getSum(void)
+        {
+            return (var1 + var2);
+        }
+
+        int getAverage(void)
+        {
+            return ((var1 + var2)/2);
+        }
+
+        int getMaxAverageValue(void)
+        {
+            return max_average_value;
+        }
+
+        bool isAverageWithinDesiredMaxValue(void) //remove void
+        {
+            if (getAverage() > getMaxAverageValue())
+            {
+                // printf("ERROR: Average outside of max_average_value. [max_average_value=%d]", max_average_value);
+                return false;
+            }
+            return true;
+        }
+
+        int getDifference(void)
```

Comment:
in c++, no need to specify void

_In C (void) means that there is no arguments required in function call, while () means unspecified number of arguments. In C++. () means the same as (void)_

Reply:
noted. void is deleted.